client/my-sites/plan-features-2023-grid/hooks/npm-ready/use-is-large-add-on-currency.ts

```diff
@@ -0,0 +1,25 @@
+import { getDisplayPrices } from '../../lib/get-display-prices';
+import type { AddOnMeta } from '@automattic/data-stores';
+
+const LARGE_CURRENCY_CHAR_THRESHOLD = 7;
+
+/**
+ * Hook that returns true if the string representation of any price of any of the storage
+ * add-ons exceeds 6 characters. For example, $4,000 would be 6 characters.
+ * This is primarily used for lowering the font-size of "large" display prices.
+ */
+export function useIsLargeAddOnCurrency( {
```

Comment:
**Note:**

There are several similarities between this hook and the `useIsLargeCurrency` hook, but I opted to create a second hook because: 

- There are several distinctions like different large character thresholds and different conditions to qualify as a large currency
- To avoid a premature abstraction

With that being said, I'm happy to combine both into an abstraction if the team prefers it.

Reply:
Thanks for outlining your thought process and stating pros and cons. 

From my perspective, it shows that the large currency concept can't be derived from either plans or add-ons, it should be derived directly from a list of prices, and should be configurable to fit the different needs we are aware of now, e.g. the thresholds. So let's combine them :)